sklearn/impute/tests/test_impute.py

```diff
@@ -1119,6 +1132,9 @@ def test_missing_indicator_error(X_fit, X_trans, params, msg_err):
 def test_missing_indicator_new(
     missing_values, arr_type, dtype, param_features, n_features, features_indices
 ):
+    if missing_values == 0 and arr_type is not np.array:
```

Comment:
Even though this if condition does seem a bit awkward, I think it looks okay. Let's see if another maintainer has a better idea.